### PR TITLE
feature: added support for checking out whole repositories

### PR DESCRIPTION
**Pull request by ireznice**

### CODE REVIEW

File: spec/integration/package_packer_spec.rb [hunk 1]
```diff
@@ -59,6 +59,32 @@
...
+      zip = Zip::File.open(package_path + '/package.zip')
+
+      expect(zip.find_entry('everything/docs/test.md')).not_to be nil
+      expect(zip.find_entry('everything/docs')).not_to be nil
+      expect(zip.find_entry('everything/lib')).not_to be nil
+      expect(zip.find_entry('everything/lib/sublib')).not_to be nil
+      expect(zip.find_entry('everything/log')).not_to be nil
+    end
+
+    it 'adds everything to final archive to root directory' do
```
Comment (ondrej-hosak) [inline spec/integration/package_packer_spec.rb]: it 'copies the whole repository and preserve repository structure'

File: spec/integration/package_packer_spec.rb [hunk 2]
```diff
@@ -59,6 +59,32 @@
 
       expect(zip.find_entry('app.rb')).not_to be nil
     end
+
+    it 'adds everything to final archive to specified directory' do
```
Comment (ondrej-hosak) [inline spec/integration/package_packer_spec.rb]: it 'copies the whole repository into destination'

**Review by ondrej-hosak — COMMENTED** [review]
Please add note to readme.md that now we are able to cope the whole repository with sample

File: lib/package_provider/zip_file_generator.rb [hunk 3]
```diff
@@ -29,6 +29,7 @@ def add_folder(input_dir, src, dest)
     def write
       ::Zip::File.open(@output_file, ::Zip::File::CREATE) do |io|
         @folders.each do |input_dir, src, dest|
+          src = '' if src == '/**' # the whole src directory could be specified by /**
```
Comment (petrSchreiber) [inline lib/package_provider/zip_file_generator.rb]: src.clear if ...

Not merge block, just seems more noble .)

File: spec/integration/package_packer_spec.rb [hunk 4]
```diff
@@ -59,6 +59,32 @@
...
+      subject.add_folder(test_path, PackageProvider::FolderOverride.new('/**', 'everything'))
+      subject.flush
+
+      zip = Zip::File.open(package_path + '/package.zip')
+
+      expect(zip.find_entry('everything/docs/test.md')).not_to be nil
```
Comment (petrSchreiber) [inline spec/integration/package_packer_spec.rb]: Shakespearean block of asserts, isn't it?

Comment (petrSchreiber) [inline spec/integration/package_packer_spec.rb]: I am commenting on the "not to be", not on the scale :)

**Pull request merged**